app/src/main/java/com/babylon/wallet/android/presentation/settings/preferences/gateways/GatewaysScreen.kt

```diff
@@ -389,38 +423,99 @@ private fun GatewayCard(
 }
 
 @Composable
-private fun Gateway.displayName(): String = string
+private fun GatewaysViewModel.State.GatewayUiItem.name(): String = when (gateway.network.id) {
+    NetworkId.MAINNET -> stringResource(id = R.string.gateway_mainnet_title)
+    NetworkId.STOKENET -> stringResource(id = R.string.gateway_stokenet_title)
+    else -> url
+}
```

Comment:
I am not sure if this is the correct approach here. We sometimes have deployed gateways with the same network id but different URL. I believe the correct way to do this is to match the official GWs with their actual identifiers, meaning their URLs.

Reply:
@sergiupuhalschi-rdx I would suggest syncing with the iOS team on this one.

Reply:
Are you saying a GW's network can have `NetworkId.MAINNET` or `NetworkId.STOKENET` but with different URLs than the official ones?